File: frontend/lib/api/outreach.ts
// Copywriter / Outreach — gerar e-mail, rascunhos em massa, follow-ups, sync.

import { request } from './client';
import type {
  CopywriterRequest,
  CopywriterResponse,
  GerarRascunhosResponse,
  GerarFollowupsRequest,
  GerarFollowupsResponse,
  SyncResponse,
  EmailHistoryResponse,
} from '../types';

export const outreachApi = {
  copywriterGerar(
    body: CopywriterRequest,
    opts?: { signal?: AbortSignal },
  ): Promise<CopywriterResponse> {
    return request<CopywriterResponse>('/api/agents/copywriter/gerar', {
      method: 'POST',
      body,
      timeoutMs: 120_000,
      signal: opts?.signal,
    });
  },

  outreachGerar(
    body: { limit?: number | null; pausa?: number },
    opts?: { signal?: AbortSignal },
  ): Promise<GerarRascunhosResponse> {
    return request<GerarRascunhosResponse>('/api/agents/outreach/gerar', {
      method: 'POST',
      body,
      timeoutMs: 600_000,
      signal: opts?.signal,
    });
  },

  outreachFollowups(
    body: GerarFollowupsRequest,
    opts?: { signal?: AbortSignal },
  ): Promise<GerarFollowupsResponse> {
    return request<GerarFollowupsResponse>('/api/agents/outreach/followups', {
      method: 'POST',
      body,
      timeoutMs: 600_000,
      signal: opts?.signal,
    });
  },

  outreachSync(opts?: { signal?: AbortSignal }): Promise<SyncResponse> {
    return request<SyncResponse>('/api/agents/outreach/sync', {
      method: 'POST',
      timeoutMs: 120_000,
      signal: opts?.signal,
    });
  },

  outreachEmails(limit: number = 50): Promise<EmailHistoryResponse> {
    return request<EmailHistoryResponse>(
      `/api/agents/outreach/emails?limit=${limit}`,
      { timeoutMs: 15_000 },
    );
  },
};

File: backend/app/api/routers/outreach.py
from fastapi import APIRouter, HTTPException

from app.api.schemas.outreach import (
    EmailHistoryResponse,
    GerarRascunhosRequest,
    GerarRascunhosResponse,
    GerarFollowupsRequest,
    GerarFollowupsResponse,
    SyncResponse,
)
from app.api.services import outreach_service

router = APIRouter(prefix="/api/agents/outreach", tags=["outreach"])


@router.post("/gerar", response_model=GerarRascunhosResponse,
             summary="Gera rascunhos pros contatos pendentes")
async def gerar(body: GerarRascunhosRequest) -> GerarRascunhosResponse:
    try:
        r = await outreach_service.gerar(limit=body.limit, pausa=body.pausa)
        return GerarRascunhosResponse(**r)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"{type(e).__name__}: {e}")

@router.post("/followups", response_model=GerarFollowupsResponse,
             summary="Gera rascunhos de follow-up pros e-mails sem resposta")
async def followups(body: GerarFollowupsRequest) -> GerarFollowupsResponse:
    try:
        r = await outreach_service.gerar_followups(
            dias=body.dias,
            max_followups=body.max_followups,
            limit=body.limit,
            pausa=body.pausa,
        )
        return GerarFollowupsResponse(**r)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"{type(e).__name__}: {e}")


@router.post("/sync", response_model=SyncResponse,
             summary="Confirma enviados e detecta respostas")
async def sync() -> SyncResponse:
    try:
        r = await outreach_service.sincronizar_async()
        return SyncResponse(**r)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"{type(e).__name__}: {e}")


@router.get("/emails", response_model=EmailHistoryResponse,
            summary="Histórico de e-mails")
async def emails(limit: int = 50) -> EmailHistoryResponse:
    try:
        items = await outreach_service.historico(limit=limit)
        return EmailHistoryResponse(items=items, total=len(items)) #type: ignore
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"{type(e).__name__}: {e}")
